cellrank/tl/_utils.py

```diff
@@ -769,6 +769,16 @@ def _connected(c: Union[spmatrix, np.ndarray]) -> bool:
     return nx.is_connected(G)
 
 
+def _irreducible(d: Union[spmatrix, np.ndarray]) -> bool:
```

Comment:
Since you're always logging after calling this function (at least in the function in this PR), consider
moving the log calls inside this function.
If there's a call where the callee afterwards does not log, please ignore this comment.

Reply:
That's a good point, thanks.

Reply:
done.